x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/upgrade_prebuilt_rules_table/use_prebuilt_rules_upgrade_state.ts

```diff
@@ -104,11 +105,18 @@ function calcFieldsState(
 
     switch (fieldDiff.conflict) {
       case ThreeWayDiffConflict.NONE:
-        fieldsState[fieldName] = {
-          state: fieldDiff.has_update
-            ? FieldUpgradeStateEnum.NoConflict
-            : FieldUpgradeStateEnum.NoUpdate,
-        };
+        if (fieldDiff.has_update) {
```

Comment:
The logic becomes quite complex here. Do you think using `fieldDiff.diff_outcome` in `switch` instead of `fieldDiff.conflict` will make it more readable?

Reply:
Just tried refactoring it to use `fieldDiff.diff_outcome` in `switch`, but it actually became even more messy. Maybe we can take a look at it together?